### PR TITLE
Test for 2785&2787

### PR DESCRIPTION
**Pull request by jannusi**
Description
-----------

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+/*! called at the end of the whole suite */
+int suiteTearDown(int numFailures)
+{
+    return numFailures;
+}
+
+/* ==============================  Test Cases  ============================== */
+
+/*  Coverage for line 2787 where scheduler running
```
Comment (rawalexe) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c:81]: Does not follow comment guidelines provided

File: FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    {
+        xTaskCreate(vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 1, &xTaskHandles[i]);
+    }
+    xTaskHandles[0]->xTaskRunState = 1;
+
+    vTaskSuspend(xTaskHandles[0]);
+}
+
+/*  Coverage for 2785 line where taskTASK_IS_RUNNING( pxTCB ) macro 
+    is being tested inside vTaskSuspend function */
```
Comment (rawalexe) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c:99]: Does not follow comment guidelines

**Review by rawalexe — COMMENTED** [review]
Please follow the guideline provided from slack

File: FreeRTOS/Test/CMock/smp/Makefile [hunk 3]
```diff
@@ -5,12 +5,15 @@ MAKEFILE_ABSPATH     := $(abspath $(lastword $(MAKEFILE_LIST)))
...
 # SUITES lists the suites contained in subdirectories of this directory
-SUITES	+=	single_priority_no_timeslice
+
 SUITES	+=	single_priority_timeslice
 SUITES	+=	multiple_priorities_no_timeslice
 SUITES	+=	single_priority_no_timeslice_covg
 SUITES	+=	single_priority_timeslice_thread
 SUITES	+=	multiple_priorities_no_timeslice_covg
+SUITES	+=	single_priority_no_timeslice
+SUITES	+=	multiple_priorities_no_timeslice_divya
```
Comment (chinglee-iot) [inline FreeRTOS/Test/CMock/smp/Makefile:15]: Create the test group ( folder ) only when FreeRTOSConfig.h is different. The test group naming should reflect the difference in FreeRTOSConfig.h.
Can we add the test cases in multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice.c?

File: FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c (hunk 1 above)
Comment (chinglee-iot) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c:81]: Please reference the "test format and guideline" document and update the test cases.
* Comment format with doxygen.
* Implementation should verify some condition with assert.
* Implementation should use mock and variables.

File: FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    for (i = 0; i < configNUMBER_OF_CORES; i++)
+    {
+        xTaskCreate(vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 1, &xTaskHandles[i]);
+    }
+    xTaskHandles[0]->xTaskRunState = 1;
+
+    vTaskSuspend(xTaskHandles[0]);
+}
+
+/*  Coverage for 2785 line where taskTASK_IS_RUNNING( pxTCB ) macro
```
Comment (alfred2g) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c:98]: line numbers are not accurate, they could change at any addition or removal of all lines before this line in tasks.c, which require a massive update to all these references.

File: FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    vTaskStartScheduler();
+
+    /* task T0 */
+    xTaskHandles[0]->xTaskRunState = configNUMBER_OF_CORES + 1;
+    vTaskSuspend(xTaskHandles[0]);
+}
+
+/*  Coverage for 2785 line where taskTASK_IS_RUNNING( pxTCB ) macro
```
Comment (alfred2g) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c:119]: 1. here too line numbers
2. not following guidelines and doxygen formatted comments

File: FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+{
+    return numFailures;
+}
+
+/* ==============================  Test Cases  ============================== */
+
+/*  Coverage for line 2787 where scheduler running 
+    needs to be pdFalse inside vTaskSuspend function  */
+
+void test_scheduler_running_false(void)
```
Comment (alfred2g) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c:84]: the name of the test needs to be
test_API_TESTED_specific_item_being_tested

File: FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    for (i = 0; i < configNUMBER_OF_CORES; i++)
+    {
+        xTaskCreate(vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 1, &xTaskHandles[i]);
+    }
+
+    vTaskStartScheduler();
+
+    /* task T0 */
+    xTaskHandles[0]->xTaskRunState = -1;
+    vTaskSuspend(xTaskHandles[0]);
```
Comment (alfred2g) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_divya/multiple_priorities_no_timeslice_divya_utest.c:137]: need to add assertions at the end, similar to all previous testcases

**Pull request closed without merging**